src/Lib.hs

```diff
@@ -426,28 +437,51 @@ getAttachmentsFromComment comments = do
 
 -- | Get zendesk responses
 -- | Returns with maybe because it could return no response
-getZendeskResponses :: [Comment] -> [Attachment] -> TicketInfo -> App (Maybe ZendeskResponse)
+getZendeskResponses :: [Comment] -> [Attachment] -> TicketInfo -> App ZendeskResponse
 getZendeskResponses comments attachments ticketInfo
     | not (null attachments) = inspectAttachments ticketInfo attachments
-    | not (null comments)    = Just <$> responseNoLogs ticketInfo
-    | otherwise              = return Nothing
+    | not (null comments)    = responseNoLogs ticketInfo
+    | otherwise              = throwM $ InvalidTicketInfo (tiId ticketInfo)
+    -- No attachment, no comments means something is wrong with ticket itself
 
 -- | Inspect only the latest attachment. We could propagate this
 -- @Maybe@ upwards or use an @Either@ which will go hand in hand
 -- with the idea that we need to improve our exception handling.
-inspectAttachments :: TicketInfo -> [Attachment] -> App (Maybe ZendeskResponse)
-inspectAttachments ticketInfo attachments = runMaybeT $ do
+inspectAttachments :: TicketInfo -> [Attachment] -> App ZendeskResponse
+inspectAttachments ticketInfo attachments = do
 
     config          <- ask
     getAttachment   <- asksZendeskLayer zlGetAttachment
 
-    let lastAttach :: Maybe Attachment
-        lastAttach = safeHead . reverse . sort $ attachments
-
-    lastAttachment  <- MaybeT . pure $ lastAttach
-    att             <- MaybeT $ getAttachment lastAttachment
-
-    pure $ inspectAttachment config ticketInfo att
+    mAtt <- runMaybeT $ do
```

Comment:
This seems overly verbose. Can you move it around a bit to make it a bit more concise?

Reply:
I'll try it out. Thank you.